ts/Series/ArcDiagram/ArcDiagramSeries.ts

```diff
@@ -0,0 +1,603 @@
+ *
+ * */
+
+import type ArcDiagramSeriesOptions from './ArcDiagramSeriesOptions';
+import ArcDiagramPoint from './ArcDiagramPoint.js';
+
+import SankeyColumnComposition from '../Sankey/SankeyColumnComposition.js';
+import SeriesRegistry from '../../Core/Series/SeriesRegistry.js';
+const {
+    seriesTypes: {
+        sankey: SankeySeries
+    }
+} = SeriesRegistry;
+import U from '../../Core/Utilities.js';
+const {
+    extend,
+    merge,
+    pick,
+    relativeLength
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.arcdiagram
+ *
+ * @augments Highcharts.seriesTypes.sankey
+ */
+class ArcDiagramSeries extends SankeySeries {
+
+    /**
+     *  Arc diagram series is a chart drawing style in which
+     *  the vertices of the chart are positioned along a line
+     *  on the Euclidean plane and the edges are drawn as a semicircle
+     *  in one of the two half-planes delimited by the line,
+     *  or as smooth curves formed by sequences of semicircles.
+     *
+     * @sample highcharts/demo/arc-diagram/
+     *         Arc Diagram
+     *
+     * @extends      plotOptions.sankey
+     * @exclude      dataSorting
+     * @since        next
+     * @product      highcharts
+     * @requires     modules/arc-diagram
+     * @optionparent plotOptions.arcdiagram
+     */
+    public static defaultOptions: ArcDiagramSeriesOptions = merge(SankeySeries.defaultOptions, {
+        nodeShape: 'circle',
+        /**
+         * Wheter the nodes with different values should have the same size.
+         *
+         * @type    {boolean}
+         * @since next
+         * @default false
+         * @product highcharts
+         */
+        equalNodes: false
+    } as ArcDiagramSeriesOptions);
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<ArcDiagramPoint> = void 0 as any;
+
+    public options: ArcDiagramSeriesOptions = void 0 as any;
+
+    public nodeColumns: Array<SankeyColumnComposition.ArrayComposition<ArcDiagramPoint>> = void 0 as any;
+
+    public nodes: Array<ArcDiagramPoint> = void 0 as any;
+
+    public points: Array<ArcDiagramPoint> = void 0 as any;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    /**
+     * Create node columns by analyzing the nodes and the relations between
+     * incoming and outgoing links.
+     * @private
+     */
+    public createNodeColumns(): Array<SankeyColumnComposition.ArrayComposition<ArcDiagramPoint>> {
+        const series = this,
+            chart = series.chart,
+            // column needs casting, to much methods required at the same time
+            column = SankeyColumnComposition.compose([] as Array<ArcDiagramPoint>, series);
+
+        column.sankeyColumn.maxLength = chart.inverted ?
+            chart.plotHeight : chart.plotWidth;
+
+        // Get the translation factor needed for each column to fill up the
+        // plot height
+        column.sankeyColumn.getTranslationFactor = (
```

Comment:
Q: It seems right now, Sankey adds `sankeyColumn` to the point via composition. Then ArcDiagram overwrites some of the methods, right? Would it make sense to use a generic NodeColumn composition, and both, Sankey and ArcDiagram implement separate compositions, both conforming to the NodeColumn one?

Reply:
`SankeyColumn` already takes methods from both `Arc Diagram` and `Sankey`, however, if the same method is used in both series types differently, it takes it from the `Sankey` by default. It might be possible to simply move `SankeyColumn` to the other, more generic place but the code will not change much.

Reply:
Okay, thanks 👍